qa/integration-tests/src/test/java/io/camunda/zeebe/it/client/command/MigrateProcessInstanceTest.java

```diff
@@ -0,0 +1,346 @@
+
+    final long jobKey =
+        RecordingExporter.jobRecords(JobIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .withType("A")
+            .getFirst()
+            .getKey();
+
+    // deploy a new version of the process
+    final long targetProcessDefinitionKey =
+        CLIENT_RULE.deployProcess(
+            Bpmn.createExecutableProcess(processId)
+                .startEvent()
+                .serviceTask("B", a -> a.zeebeJobType("B"))
+                .endEvent()
+                .done());
+
+    // when
+    final var command =
+        CLIENT_RULE
+            .getClient()
+            .newMigrateProcessInstanceCommand(processInstanceKey)
+            .migrationPlan(targetProcessDefinitionKey)
+            .addMappingInstruction("A", "B")
+            .send();
+
+    // then
+    assertThatNoException()
+        .describedAs("Expect that migration command is not rejected")
+        .isThrownBy(command::join);
+
+    final var migratedProcessInstance =
+        RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_MIGRATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .withElementType(BpmnElementType.PROCESS)
+            .getFirst()
+            .getValue();
+
+    final var migratedTaskA =
+        RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_MIGRATED)
+            .withRecordKey(serviceTaskKey)
+            .getFirst()
+            .getValue();
+
+    final var migratedJobA =
+        RecordingExporter.jobRecords(JobIntent.MIGRATED)
+            .withRecordKey(jobKey)
+            .getFirst()
+            .getValue();
+
+    assertThat(migratedProcessInstance)
+        .describedAs("Expect that process definition key changed")
+        .hasProcessDefinitionKey(targetProcessDefinitionKey)
+        .describedAs("Expect that bpmn process id and element id stayed the same")
+        .hasBpmnProcessId(processId)
+        .hasElementId(processId)
+        .describedAs("Expect that version number did not change")
+        .hasVersion(2);
+
+    assertThat(migratedTaskA)
+        .describedAs("Expect that process definition is updated")
+        .hasProcessDefinitionKey(targetProcessDefinitionKey)
+        .hasBpmnProcessId(processId)
+        .hasVersion(2)
+        .describedAs("Expect that element id changed due to mapping")
+        .hasElementId("B");
+
+    assertThat(migratedJobA)
+        .describedAs("Expect that process definition is updated")
+        .hasProcessDefinitionKey(targetProcessDefinitionKey)
+        .hasBpmnProcessId(processId)
+        .hasProcessDefinitionVersion(2)
+        .describedAs("Expect that element id changed due to mapping")
+        .hasElementId("B")
+        .describedAs(
+            "Expect that the type did not change even though it's different in the target process."
+                + " Re-evaluation of the job type expression is not enabled for this migration")
+        .hasType("A");
```

Comment:
👍 I like that you're testing the exported records in this IT, they are part of our API. So it makes sense to test this in some detail.